core/src/main/scala/org/scalafmt/rewrite/AvoidInfix.scala

```diff
@@ -29,20 +29,25 @@ case object AvoidInfix extends Rewrite {
           TokenPatch.AddLeft(fstOpToken, ".", keepTok = true)
         )
 
-        val fstArgsToken = args.head.tokens.head
-        val lastArgsToken = args.last.tokens.last
+        val fstArgsToken = args.headOption.map(_.tokens.head)
```

Comment:
Since we're already going through the effort of wrapping everything in Option, what do you think about `.flatMap(_.tokens.headOption)`?